File: January/Week1/SoHyun/S1_17615.py
import sys
input = sys.stdin.readline

n = int(input().rstrip())
balls = input().rstrip()

red = balls.count('R')
blue = n - red

res = min(red, blue)

cnt = 1
for i in range(1, n):
    if balls[i] == balls[i-1]:
        cnt += 1
    else:
        break
    
if balls[0] == 'R':
    res = min(res, red - cnt)
else:
    res = min(res, blue - cnt)
    
cnt = 1
for i in range(n-2, -1, -1):
    if balls[i] == balls[i+1]:
        cnt += 1
    else:
        break
    
if balls[-1] == 'R':
    res = min(res, red - cnt)
else:
    res = min(res, blue - cnt)
    
print(res)

File: January/Week1/AhBeen/S1_17615.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(br.readLine());
		String balls = br.readLine();

		int red = 0;
		int blue = 0;

		for(char c : balls.toCharArray()) {
			if(c == 'R') red++;
			else blue++;
		}

		int answer = Integer.MAX_VALUE;

		answer = Math.min(answer, red - countLeft(balls, 'R'));
		answer = Math.min(answer, red - countRight(balls, 'R'));
		answer = Math.min(answer, blue - countLeft(balls, 'B'));
		answer = Math.min(answer, blue - countRight(balls, 'B'));

		System.out.println(answer);
	}

	static int countLeft(String s, char target) {
		int count = 0;
		for(char c : s.toCharArray()) {
			if(c == target) count++;
			else break;
		}
		return count;
	}

	static int countRight(String s, char target) {
		int count = 0;
		for(int i = s.length() - 1; i >= 0; i--) {
			if(s.charAt(i) == target) count++;
			else break;
		}
		return count;
	}
}